tools/check_clang_tidy.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -e
+
+fail() {
+    printf "\033[1;31merror: %s: %s\033[0m\n" ${FUNCNAME[1]} "${1:-"Unknown error"}"
+    exit 1
+}
+
+hash run-clang-tidy-10.py || fail "run-clang-tidy-10.py not found"
+cd $(git rev-parse --show-toplevel)
+[ -f build/compile_commands.json ] || fail "build/compile_commands.json not found"
+# TODO: only process files change in PR
```

Comment:
Right now iceoryx does not pass checks and overall checks running quite long on my i7 with 8 HT cores it took 5+ minutes. So there tow things to consider:


1. How to make codebase complaint?
  a. reduce number of checks and add them gradually
  b.  try to fix violations in separate PR
1. Should we run clang-tidy for changed or all files?

Reply:
One thing to consider is that we need to give the developer a chance for correct linting the code before commitring & pushing to the CI, otherwise the contributors would hammer the CI.

For that we added some weeks ago the [git-hooks ](https://github.com/eclipse-iceoryx/iceoryx/tree/master/tools/git-hooks) as automatic checks while coding.  In https://github.com/eclipse-iceoryx/iceoryx/blob/master/tools/git-hooks/pre-commit you can find a short script that already do the clang-format on the touched files. We should add something similar for clang-tidy.

In the beginning I would vote for to clang-tidy only the touched files. After some time we can then add a complete scan once the codebase is migrated. We need to consider that we have no guaranteed performance on a GitHub Runner and that they are usually limited because they are for free.

Reply:
@r7vme My approach was to first add all checks and disable checks whenever they do not make sense for us.
With this in mind we could then reduce the warnings with every PR or disable another check.

I would run clang-tidy only for the changed files and we should create an issue to fix all clang-tidy warnings.